libraries/plugins/p2p/p2p_plugin.cpp

```diff
@@ -271,73 +270,9 @@ bool p2p_plugin_impl::has_item( const graphene::net::item_id& id )
 }
 
 bool p2p_plugin_impl::handle_block( const graphene::net::block_message& blk_msg, bool sync_mode, std::vector<fc::uint160_t>& )
-{ try {
-   uint32_t head_block_num;
-
-   head_block_num = chain.db().head_block_num();
-
-   if (sync_mode)
-      fc_ilog(fc::logger::get("sync"),
-            "chain pushing sync block #${block_num} ${block_hash}, head is ${head}",
-            ("block_num", blk_msg.block.block_num())
-            ("block_hash", blk_msg.block_id)
-            ("head", head_block_num));
-   else
-      fc_ilog(fc::logger::get("sync"),
-            "chain pushing block #${block_num} ${block_hash}, head is ${head}",
-            ("block_num", blk_msg.block.block_num())
-            ("block_hash", blk_msg.block_id)
-            ("head", head_block_num));
-   if (sync_mode && blk_msg.block.block_num() % 10000 == 0)
-   {
-      ilog("Syncing Blockchain --- Got block: #${n} time: ${t}",
-         ("t",blk_msg.block.timestamp)
-         ("n", blk_msg.block.block_num()) );
-   }
-
-   time_point_sec now = fc::time_point::now();
-
-   uint64_t max_accept_time = now.sec_since_epoch();
-   max_accept_time += allow_future_time;
-   FC_ASSERT( blk_msg.block.timestamp.sec_since_epoch() <= max_accept_time );
-
-   try {
-      // TODO: in the case where this block is valid but on a fork that's too old for us to switch to,
-      // you can help the network code out by throwing a block_older_than_undo_history exception.
-      // when the net code sees that, it will stop trying to push blocks from that chain, but
-      // leave that peer connected so that they can get sync blocks from us
-      bool result = chain.db().push_block( blk_msg.block, ( block_producer | force_validate ) ? chain::database::skip_nothing : chain::database::skip_transaction_signatures );
```

Comment:
The p2p plugin should get a handle on the chain plugin in plugin_initialize. Add flag field to chain_plugin::accept_block if it does not have it. This line should call accept_block.